### PR TITLE
rp2040: use TU endpoint conventions and remove redundant variables

### PR DESCRIPTION
**Pull request by majbthrd**
The rp2040 driver currently has 32 instances of three bytes (bool in, uint8_t num, uint8_t ep_addr), each of which collectively stores only one byte of information.

TinyUSB already has functions (tu_edpt_number() and tu_edpt_dir()) that perform the function of the “num” and “in” above, and other device drivers use these.

Bringing the rp2040 into line with established TU code conventions improves readability, reduces overall (code+data) size, and saves on RAM usage.

### CODE REVIEW

**Review by liamfraser — APPROVED** [review]

File: src/portable/raspberrypi/rp2040/dcd_rp2040.c [hunk 1]
```diff
@@ -131,7 +130,7 @@ static void _hw_endpoint_init(struct hw_endpoint *ep, uint8_t ep_addr, uint16_t
     ep->transfer_type = transfer_type;
 
     // Every endpoint has a buffer control register in dpram
-    if (ep->in)
+    if (tu_edpt_dir(ep->ep_addr) == TUSB_DIR_IN)
```
Comment (hathach) [inline src/portable/raspberrypi/rp2040/dcd_rp2040.c:133]: should be just  `dir == TUSB_DIR_IN`

File: src/portable/raspberrypi/rp2040/dcd_rp2040.c [hunk 2]
```diff
@@ -143,7 +142,7 @@ static void _hw_endpoint_init(struct hw_endpoint *ep, uint8_t ep_addr, uint16_t
     // Clear existing buffer control state
     *ep->buffer_control = 0;
 
-    if (ep->num == 0)
+    if (tu_edpt_number(ep->ep_addr) == 0)
```
Comment (hathach) [inline src/portable/raspberrypi/rp2040/dcd_rp2040.c:145]: should be just `num == 0`

File: src/portable/raspberrypi/rp2040/dcd_rp2040.c [hunk 3]
```diff
@@ -102,12 +103,10 @@ static void _hw_endpoint_alloc(struct hw_endpoint *ep)
 static void _hw_endpoint_init(struct hw_endpoint *ep, uint8_t ep_addr, uint16_t wMaxPacketSize, uint8_t transfer_type)
 {
     uint8_t num = tu_edpt_number(ep_addr);
-    bool in = ep_addr & TUSB_DIR_IN_MASK;
+    tusb_dir_t dir = tu_edpt_dir(ep_addr);
```
Comment (hathach) [inline src/portable/raspberrypi/rp2040/dcd_rp2040.c:106]: having both num & dir as `cont` would be nice.

**Review by hathach — COMMENTED** [review]
only a few minor feedback, if you don't have time. I could do a friendly push to update the PR :)

**Comment by majbthrd** [comment]
Done.  Thanks @hathach!

**Comment by hathach** [comment]
> Done. Thanks @hathach!

np, look perfect now, will merge when ci complete :)

**Pull request merged**